aten/src/ATen/core/op_registration/hacky_wrapper_for_legacy_signatures.h

```diff
@@ -15,6 +15,9 @@ namespace c10 {
 namespace impl {
 
 inline c10::optional<MemoryFormat> process_memory_format(const TensorOptions& options, c10::optional<MemoryFormat> memory_format) {
+    TORCH_CHECK(options.requires_grad_opt() == c10::nullopt,
+        "Operators taking TensorOptions cannot take a TensorOptions with "
+        "options.requires_grad set as ", options.requires_grad());
```

Comment:
It's weird to do a requires grad opt test in `process_memory_format`

Reply:
True. It's unfortunately the only place where I can add it without big codegen changes. Open to ideas.

Reply:
Rename `process_memory_format` then?